File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.js
function solution(arr) {
    while(arr.length !== arr[0].length){
        if(arr.length < arr[0].length){
            arr.push(Array(arr[0].length).fill(0));
        }else{
            arr.map((val,idx) => val = val.push(0));
        }
    }
    
    return arr;
}

File: 프로그래머스/0/181830. 정사각형으로 만들기/정사각형으로 만들기.py
def solution(arr):
    answer = [[]]
    
    while(True):
        row = len(arr)
        col = len(arr[0])
    
        if row > col :
            for i in range(row):
                arr[i].append(0)
        
        elif row < col :
            arr.append([0]*col)
        
        elif row == col :
            answer = arr
            break;
    
    return answer